agent/server/session.go

```diff
@@ -533,6 +548,43 @@ func (s *session) ping6(ctx context.Context, args ...string) {
 	}
 }
 
+var errMalformedSetToken = errors.New("malformed set-token command")
+
+// setToken instructs the agent which tokens to use for API calls.
+func (s *session) setToken(ctx context.Context, args ...string) {
+	s.exactArgs(2, args, errMalformedSetToken)
+
+	switch args[0] {
+	case "cfg":
+		tokStr, err := config.ReadAccessToken(args[1])
+		if err != nil {
+			s.error(err)
+			return
+		}
+
+		s.tokens = tokens.Parse(tokStr)
+		s.tokens.FromConfigFile = args[1]
+	case "str":
+		s.tokens = tokens.Parse(args[1])
+	}
+
+	go s.srv.UpdateTokensFromClient(s.tokens)
+
+	s.ok()
+
+	s.runCommand(ctx)
```

Comment:
Each command from the client is it's own session. Since we want to set the token per-session, we have to do it on each command. So a session might look like

```
-> set-token cfg ~/.fly/config.toml
<- ok
-> ping whatever
<- ok whatever
```